lib/primitives/object-element.js

```diff
@@ -45,6 +45,14 @@ module.exports = function(BaseElement, ArrayElement, MemberElement) {
       );
     },
 
+    delete: function (name) {
+      this.content = this.content.filter(function (item) {
+        return item.key.toValue() !== name;
+      });
+
+      return this;
```

Comment:
I don't think it makes sense to return the collection. Unless it was returning a new collection with the removed item (non mutating).

```javascript
const newObject = object.removing('somekey');
```

I think its far more likely you would want to chain the deleted element itself. This remove method doesn't offer the caller any feedback on whether the element was removed from the collection.

```javascript
if (object.remove('request')) {
    console.log('there was a request and we removed it');
}
```

```javascript
const response = object.remove('response');
if (response) {
    console.log('there was a response and we removed it');
    // do something with response.statusCode
}
```